File: go-version/leetcode/LC2829/minimum_sum.go
package LC2829

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/3/26 上午8:33
 * @FilePath: minimum_sum
 * @Description: 使用贪心+等差数列
 */

func minimumSum(n int, k int) int {
	leftSum := sumArithmeticSequence(1, 1, min(n, k>>1))
	rightSum := sumArithmeticSequence(k, 1, n-(k>>1))
	if n <= (k >> 1) {
		return leftSum
	}
	return leftSum + rightSum
}

func sumArithmeticSequence(a1, d, n int) int {
	return n*a1 + ((n - 1) * n * d >> 1)
}

File: python-version/LC2829/minimum_sum.py
class Solution:
    def minimumSum(self, n: int, k: int) -> int:
        left_sum = self.__sum_arithmetic_sequence(1, 1, min(n, (k >> 1)))
        right_sum = self.__sum_arithmetic_sequence(k, 1, n - (k >> 1))
        if n <= (k >> 1):
            return left_sum
        return left_sum + right_sum

    @staticmethod
    def __sum_arithmetic_sequence(a1: int, d: int, n: int) -> int:
        return n * a1 + ((n - 1) * n * d >> 1)
